### PR TITLE
Revises CONTRIBUTING.md

### LINKED ISSUE #10053: Create CONTRIBUTING file for Flutter website
The CONTRIBUTING page needs to be updated for current practices of writing and reviewing.

### PR DESCRIPTION
**Pull request by atsansone**
Revises contributing guidelines. Fixes #10053

### CODE REVIEW

File: CONTRIBUTING.md
Comment (domesticmouse) [inline CONTRIBUTING.md:8]: line length?

Comment (atsansone) [reply CONTRIBUTING.md:8]: I thought about that, too, but this isn't a public web site file. Text wrap works fine. If it's a concern, I can fix it.

**Review by domesticmouse — APPROVED** [review]

File: CONTRIBUTING.md
Comment (MaryaBelanger) [reply CONTRIBUTING.md:8]: I think the point, or one of the points, of semantic line breaks is ease of review. I'd appreciate them

**Comment by atsansone** [comment]
@RedBrogdon , @sfshaza2 , @MaryaBelanger : PTAL.

File: CONTRIBUTING.md
Comment (sfshaza2) [inline CONTRIBUTING.md:75]: Not everyone has permission to add labels.

File: CONTRIBUTING.md
Comment (sfshaza2) [inline CONTRIBUTING.md:105]: I would expect that if a PR violates the code of conduct, that we would just close it _and_ close comments on it.

File: CONTRIBUTING.md
Comment (sfshaza2) [inline CONTRIBUTING.md:107]: Review the technical concerns? I think it's Review the technical content

File: CONTRIBUTING.md
Comment (sfshaza2) [inline CONTRIBUTING.md:153]: Again, not all external contributors can apply labels.

File: CONTRIBUTING.md
Comment (sfshaza2) [inline CONTRIBUTING.md:159]: Ditto

**Review by sfshaza2 — COMMENTED** [review]
Some feedback.

File: CONTRIBUTING.md
Comment (MaryaBelanger) [inline CONTRIBUTING.md:17]: What do they do with google doc to get the flutter team to review? Do they post the google doc link in an issue they create? Or as a comment if it's an existing issue?

File: CONTRIBUTING.md
Comment (MaryaBelanger) [inline CONTRIBUTING.md:12]: If they DO find one, what then? Do they have to announce they're working on it? Wait for approval? Just start working on it? Etc.

File: CONTRIBUTING.md
Comment (MaryaBelanger) [inline CONTRIBUTING.md:23]:
```suggestion
   The new branch should be based on `main` or a feature or version branch.
```

File: CONTRIBUTING.md
Comment (MaryaBelanger) [inline CONTRIBUTING.md:30]: Would "a new commit" still be in the same PR? The previous sentence says "as part of the PR"

File: CONTRIBUTING.md
Comment (MaryaBelanger) [inline CONTRIBUTING.md:39]:
```suggestion
   and [**Highlights**][gs-highlights]
```

File: CONTRIBUTING.md
Comment (MaryaBelanger) [inline CONTRIBUTING.md:29]: "unrelated to your PR" -- maybe link to `Submit a PR` section since, at this point, you haven't told them to create a PR yet

File: CONTRIBUTING.md
Comment (MaryaBelanger) [inline CONTRIBUTING.md:54]:
```suggestion
   your PR might be merged to a dedicated branch with a different name.
```

File: CONTRIBUTING.md
Comment (MaryaBelanger) [inline CONTRIBUTING.md:82]:
```suggestion
   Acknowledge all comments with at least a [reaction emoji][reaction-emoji].
```

File: CONTRIBUTING.md
Comment (MaryaBelanger) [inline CONTRIBUTING.md:97]:
```suggestion
* Technical review comes before content, or "copy", review.
```

File: CONTRIBUTING.md
Comment (MaryaBelanger) [inline CONTRIBUTING.md:113]: "technically correct but could be done differently" -- not sure what the idea behind this statement is. Like, you could do something technically correctly in a long, convoluted, messy way. Best practices should be enforced if the reviewer has that context

File: CONTRIBUTING.md
Comment (MaryaBelanger) [inline CONTRIBUTING.md:114]:
```suggestion
* Provide suggested changes with corrected copy to make your changes clear.
```

File: CONTRIBUTING.md
Comment (MaryaBelanger) [inline CONTRIBUTING.md:161]: I think these headings should be more clear. It's not clear from the headings or the contents in each section that these are two separate types of reviews, and if you're requested as a reviewer, you're likely only responsible for one (correct?) Something like

```suggestion
### Copy review

If you're requested for a _copy_ review... (follow these practices)
```

**Review by MaryaBelanger — COMMENTED** [review]
Some initial comments. Also, not sure how it renders, but each numbered item in the lists is just a `1.`

File: CONTRIBUTING.md
Comment (atsansone) [reply CONTRIBUTING.md:153]: Anyone with triage access could, which I believe all reviewers can do. I'm double-checking with @godofredoc .

File: CONTRIBUTING.md
Comment (atsansone) [reply CONTRIBUTING.md:161]: That's tricky. I'm trying to keep headings parallel.

File: CONTRIBUTING.md
Comment (atsansone) [reply CONTRIBUTING.md:159]: Understood. Checking this.

**Review by sfshaza2 — COMMENTED** [review]
Why does this PR affect close to 60 files? It shouldn't affect much more than 1.

**Comment by atsansone** [comment]
Bad rebase. I'll fix it.

**Review by RedBrogdon — APPROVED** [review]
LGTM once @MaryaBelanger and @sfshaza2 sign off. I'm really glad to see y'all arriving at the right set of guidelines!

**Comment by sfshaza2** [comment]
@atsansone, can you address @MaryaBelanger's concerns?

File: CONTRIBUTING.md
Comment (johnpryan) [inline CONTRIBUTING.md:59]: I'm not sure what this means. Aren't contributors required to use git? What's the alternative?

File: CONTRIBUTING.md
Comment (johnpryan) [inline CONTRIBUTING.md:131]: Why is this worth calling out? How are technical comments treated differently from other comments?

File: CONTRIBUTING.md
Comment (johnpryan) [inline CONTRIBUTING.md:136]: This is a lot for me to think about. Shouldn't I just review a PR if I'm asked to review it?

File: CONTRIBUTING.md
Comment (johnpryan) [inline CONTRIBUTING.md:160]: I usually use "nit" or "nitpick" to label a comment as a nice-to-have.

File: CONTRIBUTING.md
Comment (johnpryan) [inline CONTRIBUTING.md:185]: Does the writer refer to the TW or the author of the PR?

Comment (atsansone) [reply CONTRIBUTING.md:185]: Author. I can clarify that.

File: CONTRIBUTING.md
Comment (johnpryan) [inline CONTRIBUTING.md:179]: What's the purpose of distinguishing between these two types of reviews? There's a lot of redundant copy in these two sections, maybe we should de-dupe the instructions a bit.

File: CONTRIBUTING.md
Comment (atsansone) [reply CONTRIBUTING.md:136]: Right now, GitHub actions set who reviews. The hope will be to remove this automatic setting soon. Because of this automation, a review for a PR is requested even if the author is not done with it. The review request via GitHub has lost its meaning as is.

File: CONTRIBUTING.md
Comment (johnpryan) [inline CONTRIBUTING.md:230]: We should reference official docs as much as possible: https://docs.github.com/en/issues/tracking-your-work-with-issues/filtering-and-searching-issues-and-pull-requests

File: CONTRIBUTING.md
Comment (atsansone) [reply CONTRIBUTING.md:160]: I had a whole thing in here about [conventional comments](https://conventionalcomments.org) to set a standard, but it was deemed too complex. You can do as you describe here, of course.

File: CONTRIBUTING.md
Comment (johnpryan) [inline CONTRIBUTING.md:48]: I don't think the reader is ready to write a PR summary, since they haven't pushed their code yet. You are already giving instructions for this in the "Write the PR summary using the following guidelines" section on line 72. Unless you are talking about the commit message?

File: CONTRIBUTING.md
Comment (johnpryan) [inline CONTRIBUTING.md:28]: It doesn't make sense that we would tell the reader what to name their branch. Instead, I would much rather we ask them to use the "fixes #number" syntax as it makes issue management easier: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue

File: CONTRIBUTING.md
Comment (johnpryan) [reply CONTRIBUTING.md:136]: It would be GREAT if we could only add the DRE that is on rotation. The original plan was to use a weekly rotation to reduce the email noise for everyone.

**Comment by atsansone** [comment]
@sfshaza2 , @MaryaBelanger : PTAL.

File: CONTRIBUTING.md
Comment (khanhnwin) [reply CONTRIBUTING.md:140]: The framework team individually tags reviewers. Maybe we can copy that workflow here for the DRE who's on rotation that week?

File: CONTRIBUTING.md
Comment (atsansone) [reply CONTRIBUTING.md:31]: This is a suggestion, not a mandate. It gives them direction in case they had no other ideas.

File: CONTRIBUTING.md
Comment (atsansone) [reply CONTRIBUTING.md:183]: The appropriate reviewer can key in on one section rather than going back and forth across the page.

**Comment by sfshaza2** [comment]
Hi Tony. Given that last week's meeting resulting a decision to substantively change this PR, I'm going to close it. If you'd like me to work on a new PR, let me know. Otherwise, please open a new PR for the new contributing guide.

**Pull request closed without merging**
